File: app/src/main/java/com/tentlers/mngapp/data/tables/bills/BillEntryTypeObject.java
package com.tentlers.mngapp.data.tables.bills;

public class BillEntryTypeObject {

    public BillEntryTypeObject setTenantId(long tenantId) {
        this.tenantId = tenantId;
        this.isBillSpecificTenant = true;
        return this;
    }

    public BillEntryTypeObject setRoomId(int roomId) {
        this.roomId = roomId;
        this.isBillSpecificRoom = true;
        return this;
    }

    public long tenantId;
    public int roomId;
    public boolean isBillSpecificTenant;
    public boolean isBillSpecificRoom;

    public BillEntryTypeObject setBillNormalPaid(boolean billNormalPaid) {
        isBillNormalPaid = billNormalPaid;
        return this;
    }

    public boolean isBillNormalPaid;
}

File: app/src/main/java/com/tentlers/mngapp/data/tables/bills/BillEntryTypeObject.kt
package com.tentlers.mngapp.data.tables.bills

class BillEntryTypeObject {
    fun setTenantId(tenantId: Long): BillEntryTypeObject {
        this.tenantId = tenantId
        isBillSpecificTenant = true
        return this
    }

    fun setRoomId(roomId: Int): BillEntryTypeObject {
        this.roomId = roomId
        isBillSpecificRoom = true
        return this
    }

    @JvmField
    var tenantId: Long = 0
    @JvmField
    var roomId = 0
    @JvmField
    var isBillSpecificTenant = false
    @JvmField
    var isBillSpecificRoom = false
    fun setBillNormalPaid(billNormalPaid: Boolean): BillEntryTypeObject {
        isBillNormalPaid = billNormalPaid
        return this
    }

    @JvmField
    var isBillNormalPaid = false
}